evm/chains/sharding/mainchain_handler/mainchain_handler.py

```diff
@@ -0,0 +1,78 @@
+import time
+
+import rlp
+
+from eth_utils import (
+    to_checksum_address,
+)
+
+from evm.chains.sharding.mainchain_handler.config import (
+    GASPRICE,
+    TX_GAS,
+)
+
+
+class MainchainHandler:
+
+    def __init__(self, web3_instance):
+        self.w3 = web3_instance
+        assert self.w3.isConnected()
+
+    # RPC related
+
+    def get_block_by_number(self, block_number):
+        return self.w3.eth.getBlock(block_number)
+
+    def get_block_number(self):
+        return self.w3.eth.blockNumber
+
+    def get_code(self, address):
+        address = to_checksum_address(address)
+        return self.w3.eth.getCode(address)
+
+    def get_nonce(self, address):
+        address = to_checksum_address(address)
+        return self.w3.eth.getTransactionCount(address)
+
+    def mine(self, number):
+        expected_block_number = self.get_block_number() + number
+        self.w3.miner.start(1)
+        while self.get_block_number() < expected_block_number:
+            time.sleep(0.1)
+        self.w3.miner.stop()
+
+    def unlock_account(self, account, passphrase):
```

Comment:
Your latest comment suggested this was being removed.

Reply:
sorry for that. I have removed it.